core/modules/node/node.module

```diff
@@ -2293,10 +2293,10 @@ function node_search_validate($form, &$form_state) {
   if (isset($form_state['values']['term']) && is_array($form_state['values']['term']) && count($form_state['values']['term'])) {
     $keys = search_expression_insert($keys, 'term', implode(',', $form_state['values']['term']));
   }
-  if (isset($form_state['values']['language']) && is_array($form_state['values']['language'])) {
-    $languages = array_filter($form_state['values']['language']);
+  if (isset($form_state['values']['langcode']) && is_array($form_state['values']['langcode'])) {
+    $languages = array_filter($form_state['values']['langcode']);
     if (count($languages)) {
-      $keys = search_expression_insert($keys, 'language', implode(',', $languages));
+      $keys = search_expression_insert($keys, 'langcode', implode(',', $languages));
```

Comment:
Could we change *only* this line and leave all the others as before? That would maintain compatibility in the search-box query strings.